### PR TITLE
Update programmer tcl help text

### PR DESCRIPTION
**Pull request by guanyung-rs**
> ### Motivate of the pull request
> - [x] To address an existing issue. If so, please provide a link to the issue: <issue id>
> - [ ] Breaking new feature. If so, please describe details in the description part.

Update programmer tcl command help text.

> ### Describe the technical details

> #### What is currently done? (Provide issue link if applicable)
> 
> 
> 
> 
> 
>
> #### What does this pull request change?
> 
> 
> 
> 
> 

> ### Which part of the code base require a change
> 
> - [ ] Library: <Specify the library name>
> - [ ] Plug-in: <Specify the plugin name>
> - [ ] Engine
> - [x] Documentation
> - [ ] Regression tests
> - [ ] Continous Integration (CI) scripts

> ### Impact of the pull request

> - [ ] Require a change on Quality of Results (QoR)
> - [ ] Break back-compatibility. If so, please list who may be influenced.

### CODE REVIEW

**Comment by guanyung-rs** [comment]
If these update is fine, I will update these help text to upstream repo

File: etc/help.txt [hunk 1]
```diff
@@ -195,15 +195,26 @@ Tcl commands (Available in GUI or Batch console or Batch script):
                               : Uses alternate bitstream generation configuration files
 </openfpga>
 
----------------
---- Program ---
----------------
+------------------
+--- Programmer ---
+------------------
    programmer <command>
```
Comment (ravic-rs) [inline etc/help.txt:201]: Please add a short description for "programmer" tcl command like others.
e.g. _add_litex_ip_catalog <directory> : Browses directory for LiteX IP generators, adds the IP(s) to the IP Catalog_

**Review by ravic-rs — CHANGES_REQUESTED** [review]
1 comment to address.

File: etc/help.txt [hunk 2]
```diff
@@ -195,15 +195,26 @@ Tcl commands (Available in GUI or Batch console or Batch script):
...
-     list_devices                 : List all available devices
+     command                             :fpga_config, flash, fpga_status, list_device, list_cable
+     fpga_config -c <cable name/index> ?-d <device_index>? <bitstream_file> : Program the FPGA device via JTAG
+         -c <cable_name/index>    : Specify the cable name or index
+         -d <device_index>        : Specify the device index
+         <bitstream_file>         : Bitstream file to program
+     flash -c <cable name/index> ?-d <device_index>? ?-o <operations>? <bitstream_file> : Program the Flash via JTAG
+         -c <cable_name/index>    : Specify the cable name or index
+         -d <device_index>        : Specify the device index
+         -o <operations>          : Specify the operations to perform on the flash (erase|blankcheck|program|verify)
```
Comment (esther-rs) [inline etc/help.txt:210]: Please remove the flash operations that are not supported yet.

**Review by esther-rs — CHANGES_REQUESTED** [review]

File: etc/help.txt (hunk 1 above)
Comment (guanyung-rs) [reply etc/help.txt]: Added short description.

File: etc/help.txt (hunk 2 above)
Comment (guanyung-rs) [reply etc/help.txt]: Removed the unsupported operations help

**Review by esther-rs — APPROVED** [review]

File: etc/help.txt [hunk 3]
```diff
@@ -195,15 +195,26 @@ Tcl commands (Available in GUI or Batch console or Batch script):
...
-     flash <bitstream_file>       : Program the flash via JTAG
-     fpga_status                  : Get the status of the device
-     list_devices                 : List all available devices
+------------------
+--- Programmer ---
+------------------
+   programmer <command>           : A programmer tool allows you to program and configure FPGA devices via JTAG 
+     command                      : fpga_config, flash, fpga_status, list_device, list_cable
+     fpga_config -c <cable name/index> ?-d <device_index>? <bitstream_file> : Program the FPGA device via JTAG
+         -c <cable_name/index>    : Specify the cable name or index
```
Comment (KochynVolodymyr) [inline etc/help.txt:204]: Please make shift by 2 spaces here but not 4. Same for all other keys on this level

**Review by chernyee-rs — APPROVED** [review]

File: etc/help.txt (hunk 3 above)
Comment (guanyung-rs) [reply etc/help.txt]: Updated them to two spaces

**Review by ravic-rs — APPROVED** [review]

**Comment by ravic-rs** [comment]
@guanyung-rs please go ahead and add these to upstream as well. For engineering/internal only commands, there's a tag, just FYI.

**Pull request merged**